src/login/connect_server.h

```diff
@@ -0,0 +1,2257 @@
+    UNUSED_EXPANSION_2      = 0x2000,
+    UNUSED_EXPANSION_3      = 0x4000,
+    UNUSED_EXPANSION_4      = 0x8000,
+};
+
+// Displays features on main menu // May be a 32 bit integer on the client.
+enum FEATURE_DISPLAY : uint16
+{
+    SECURE_TOKEN     = 0x0001,
+    UNUSED_FEATURE_1 = 0x0002,
+    MOG_WARDROBE_3   = 0x0004,
+    MOG_WARDROBE_4   = 0x0008,
+    MOG_WARDROBE_5   = 0x0010,
+    MOG_WARDROBE_6   = 0x0020,
+    MOG_WARDROBE_7   = 0x0040,
+    MOG_WARDROBE_8   = 0x0080,
+    UNUSED_FEATURE_2 = 0x0100,
+    UNUSED_FEATURE_3 = 0x0200,
+    UNUSED_FEATURE_4 = 0x0400,
+    UNUSED_FEATURE_5 = 0x0800,
+    UNUSED_FEATURE_6 = 0x1000,
+    UNUSED_FEATURE_7 = 0x2000,
+    UNUSED_FEATURE_8 = 0x4000,
+    UNUSED_FEATURE_9 = 0x8000,
+};
+
+// Metadata about each session
+struct session_t
+{
+    // Auth session is not yet needed to be stored
+    // std::shared_ptr<handler_session> auth_session;
+    std::shared_ptr<handler_session> data_session;
+    std::shared_ptr<handler_session> view_session;
+    std::shared_ptr<handler_session> pol_session;
+
+    uint32      accountID                 = 0;
+    uint32      serverIP                  = 0;
+    uint32      requestedCharacterID      = 0;
+    std::string requestedNewCharacterName = "";
+    bool        justCreatedNewChar        = false;
+    bool        versionMismatch           = false;
+
+    std::chrono::time_point<std::chrono::system_clock> authorizedTime = server_clock::now();
+};
+
+// NOTE: This collection of flags is 64-bits wide!
+enum AUTH_COMPONENTS
+{
+    // Information to be sent during login sequence.
+    // Each item is added to the data sequentially,
+    // after username and password, which are mandatory:
+    // [username][password][email]
+    // If you opted not to use email:
+    // [username][password]
+    SEND_EMAIL       = 1 << 0,
+    SEND_HOSTNAME    = 1 << 1,
+    SEND_MAC_ADDRESS = 1 << 2,
+
+    // These flags enable additional workflows between connect and xiloader
+    ENABLE_ACCOUNT_CREATE  = 1 << 3, // Ability for a user to create an account
+    ENABLE_ACCOUNT_DELETE  = 1 << 4, // Ability for a user to delete their account
+    ENABLE_PASSWORD_CHANGE = 1 << 5, // Ability for a user to change their password through xiloader
+    ENABLE_PASSWORD_RESET  = 1 << 6, // Ability for a user to flag their account for a password reset through connect server
+};
+
+// [ip_addr][session_hash] = session
+std::unordered_map<std::string, std::map<std::string, session_t>> authenticatedSessions_;
+
+session_t& get_authenticated_session(std::string ipAddr, std::string sessionHash)
+{
+    return authenticatedSessions_[ipAddr][sessionHash]; // NOTE: Will construct if doesn't exist
+}
+
+// hostname/ip conversion functions
+std::string ip2str(uint32 ip)
+{
+    uint32 reversed_ip = htonl(ip);
+    char   address[INET_ADDRSTRLEN];
+    inet_ntop(AF_INET, &reversed_ip, address, INET_ADDRSTRLEN);
+    return fmt::format("{}", address);
+}
+
+uint32 str2ip(const char* ip_str)
```

Comment:
All of these "global" helper functions used just in here should be in an anonymous namespace in the cpp file for connect_server

Reply:
Done, but I moved it to a helper file because many classes access those "global" functions